pkg/tracker/index_test.go

```diff
@@ -3,40 +3,38 @@
 
 package tracker
 
-// import (
-// 	"fmt"
-// 	"testing"
+import (
+	"fmt"
+	"testing"
 
-// 	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/tellor-io/telliot/pkg/tcontext"
+	"github.com/tellor-io/telliot/pkg/testutil"
+)
 
-// 	"github.com/tellor-io/telliot/pkg/common"
-// 	"github.com/tellor-io/telliot/pkg/db"
-// 	"github.com/tellor-io/telliot/pkg/testutil"
-// )
-// func TestPSR(t *testing.T) {
-// 	ctx, _, cleanup := tcontext.CreateTestContext(t)
-// 	t.Cleanup(cleanup)
-// 	psr, err := BuildIndexTrackers()
-// 	if err != nil {
-// 		testutil.Ok(t, err)
-// 	}
-// 	for idx := range psr {
-// 		err = psr[idx].Exec(ctx)
-// 		psrStr := psr[idx].String()
-// 		if err != nil {
-// 			testutil.Ok(t, errors.New(fmt.SprintF("failed to execute psr: %s %v", psrStr, err)))
-// 		}
-// 	}
-// 	val, err := ctx.Value(common.DBContextKey).(db.DB).Get(fmt.Sprintf("qv_%d", 1))
-// 	if err != nil {
-// 		testutil.Ok(t, err)
-// 	}
-// 	if val == nil {
-// 		testutil.Ok(t, errors.New(Expected a value stored for request ID 1))
-// 	}
-// 	intVal, err := hexutil.DecodeBig(string(val))
-// 	if err != nil {
-// 		testutil.Ok(t, err)
-// 	}
-// 	fmt.Println("DB value", intVal)
-// }
+func TestPSR(t *testing.T) {
```

Comment:
Actually, we disabled this one because it was flaky. Some APIs have rate limits.

Instead of this, I imagine the following:

Move the config parsing into a separate function,
Parse the config and build an array `[]IndexTracker` for `source` in `IndexTracker` can have some mock `DataSource` that returns something we can test.

Wi

And then can call each tracker in the array `i.DataSource.Get` with the result can call `i.parsePayload(results)` and check expected vs actual result.

Reply:
Perfect, I felt like we need to make this code more testable. this seems like a good idea, I'm working on it.